data/basics/userBalance.go

```diff
@@ -35,6 +35,17 @@ const (
 	Online
 	// NotParticipating indicates that the associated account is neither a delegator nor a delegate. Currently it is reserved for the incentive pool.
 	NotParticipating
+
+	// MaxEncodedAccountDataSize is a rough estimate for the worst-case scenario we're going to have of the account data and address serialized.
+	// this number is verified by the TestEncodedAccountDataSize function.
+	MaxEncodedAccountDataSize = 324205
```

Comment:
I think you should add some buffer to this in case it's off a bit because of e.g. variable integer size encoding.

Reply:
(I can see you tried to ensure the integers were all the maximum size -- but some extra headroom seems like a good idea because the downside is quite bad)